Use product name or title in products report

tomalec opened the pull request:
_Requires https://github.com/woocommerce/google-listings-and-ads/pull/669_

### Changes proposed in this Pull Request:

Use product name or title in products report, 
as provided by the API.

Use `name` if given by the Google API  for Ads, or `title` that contains title fetched from local DB by our API for Free listings.

Fixes #585


### Screenshots:

![Screencast showing product titles shown in compare table for the free and paid campaign, that matches the name and title given in the API response](https://user-images.githubusercontent.com/17435/119876604-5aecd680-bf28-11eb-960a-718be650590a.gif)



### Detailed test instructions:

0. Make sure you have some test data, like in https://github.com/woocommerce/google-listings-and-ads/pull/659
1. Go to Products Report [`/wp-admin/admin.php?page=wc-admin&path=/google/reports/products`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=/google/reports/products)
2. Check the product titles matches the names served by our API
3. Switch to "Show data from [Free listings]"
4. Check the product titles matches the ones served by our API


### Changelog Note:

> Fix - Render product titles on the products report page as given by the API.

### Additional notes:

We render `name | title` as given by the API, so if we fail to find a matching product in local DB, we will show its id (`"gla_123"`), as not "Product #123" as before. See https://github.com/woocommerce/google-listings-and-ads/pull/669#pullrequestreview-670442520

tomalec committed:
Use product name or title in products report,

as provided by the API.
Use `name` if given by the Google API  for Ads, or `title` that contains title fetched from local DB by our API for Free listings.

Fixes https://github.com/woocommerce/google-listings-and-ads/issues/585

mikkamp reviewed (approved):
Looks great, I can see the titles appear for Free Listings.

![image](https://user-images.githubusercontent.com/11388669/119936234-be0a5780-bf80-11eb-9f77-eeff5875cfeb.png)

eason9487 reviewed (approved):
LGTM.

tomalec committed:
Use `.name` to render product title.

as it was unified in https://github.com/woocommerce/google-listings-and-ads/pull/669/commits/cf5e6c82ebd9409ea479f24b8d245d3a987e2596

Co-authored-by: Mik <mikkamp@users.noreply.github.com>
